src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.cs

```diff
@@ -0,0 +1,956 @@
+                            }
+
+                            if (attribute.Obsoleted != null)
+                            {
+                                if (attribute.SupportedSecond != null && attribute.SupportedSecond > attribute.Obsoleted || attribute.SupportedFirst > attribute.Obsoleted)
+                                {
+                                    // Can supported version be greater than obsoleted? Do we want to report diagnostic here for wrong version?
+                                }
+                                else if (!SuppresedByUnsupported(callSiteAttribute, attribute.Obsoleted) && !ObsoletedSuppressed(callSiteAttribute.Obsoleted, attribute.Obsoleted))
+                                {
+                                    diagnosticAttribute.Obsoleted = (Version)attribute.Obsoleted.Clone();
+                                }
+                            }
+                        }
+                    }
+                    else if (attribute.UnsupportedFirst != null) // also means Unsupported < Supported, optional list
+                    {
+                        if (supportedOnlyList.HasValue && supportedOnlyList.Value)
+                        {
+                            // report inconsistent list diagnostic
+                            return true; // do not need to add this API to the list
+                        }
+                        else
+                        {
+                            supportedOnlyList = false;
+                        }
+
+                        if (callSiteAttributes.TryGetValue(key, out var callSiteAttribute))
+                        {
+                            if (!OptionalOsVersionsSuppressed(callSiteAttribute, attribute))
+                            {
+                                diagnosticAttribute.SupportedFirst = (Version)attribute.SupportedFirst.Clone();
+                            }
+
+                            if (!UnsupportedFirstSuppressed(attribute, callSiteAttribute))
+                            {
+                                diagnosticAttribute.UnsupportedFirst = (Version)attribute.UnsupportedFirst.Clone();
+                            }
+
+                            if (attribute.UnsupportedSecond != null && !UnsupportedSecondSuppressed(attribute, callSiteAttribute))
+                            {
+                                diagnosticAttribute.UnsupportedSecond = (Version)attribute.UnsupportedSecond.Clone();
+                            }
+
+                            if (attribute.Obsoleted != null)
+                            {
+                                if (attribute.SupportedSecond != null && attribute.SupportedSecond > attribute.Obsoleted || attribute.SupportedFirst > attribute.Obsoleted)
+                                {
+                                    // Can supported version be greater than obsoleted? Do we want to report diagnostic here for wrong version?
```

Comment:
I believe that's a declaration site error we should cover as part of #4022.

Reply:
yep